R/quantifyTranscriptionFactorSimilarity.R

```diff
@@ -0,0 +1,152 @@
+# quantifyTranscriptionFactorSimilarity.R
+#
+# Purpose: To quantify similarity between TF co-occurence and absence
+# Version: 1.0
+# Date: 2019-03-25
+# Author: Rachel Woo
+# License: MIT
+#
+# Input: An excel spreadsheet of the system parsed using the BCB420-2019-resources scripts.
+# Output: Dengrogram of TF  co-occurance written to
+#         ~/inst/img/TFOccuranceDendrogram.jpg
+# Dependencies: xlsx, tidyverse, readxl
+# ==============================================================================
+
+# SIDE EFFECTS:
+# Displays dendrogram
+
+# # ====  PACKAGES  ==============================================================
+# # Get packages
+# if (requireNamespace("xlsx", quietly=TRUE)) {
+#   install.packages("tidyverse", repos = "http://cran.us.r-project.org")
+# }
+# if (requireNamespace("devtools", quietly=TRUE)) {
+#   install.packages("devtools", repos = "http://cran.us.r-project.org")
+#   install.packages("Biostrings", repos = "http://cran.us.r-project.org")
+#   install.packages("Rtools", repos = "http://cran.us.r-project.org")
+# }
+#
+#
+# # Load packages
+# require(readxl, quietly = TRUE)
+
+
+# ==== FUNCTIONS ===============================================================
+
+
+#' Helper Function: Read system data from excel
+#'
+#' @param filepath to system excel sheet
+#' Helper function to read in system data from excel
+#' @return sysHGNC. A list of genes in the system
+#' @examples
+#' \donttest{
+#' getSysGenes("BCB420-2019-System-PHALY-0.3.xlsx")
+#' }
+getSysGenes <- function(filepath) {
+  # 1. Receive input data from excel spreadsheet and extract gene names with package tools
+  sysComponent <- readxl::read_excel(filepath, sheet = "component", skip = 1)
+  sysHGNC <- sysComponent$sym[sysComponent$molType == "protein"]
+  #Get rid of NA
+  sysHGNC <- sysHGNC[!is.na(sysHGNC)]
+  return(sysHGNC)
+}
+
+#' Helper Function: get TF data only for genes in system
+#'
+#' @param sysHGNC list of genes from system
+#' Accesses loaded TF data and filters for genes present in system
+#' @return sysTF. A smaller TF geneList with only genes in system.
+#' @examples
+#' \donttest{
+#' getSysTFdata(sysHGNC)
+#' }
+getSysTFdata <- function(sysHGNC) {
+   # From Steipe, 2019 ESA readMe
+   myURL <- paste0("http://steipe.biochemistry.utoronto.ca/abc/assets/",
+                  "geneList-2019-03-13.RData")
+  load(url(myURL))  # loads GTRD geneList object
+  # Only take elements from loaded data relevant to system
+  sysTF <- geneList[sysHGNC]
+  sysTF <- sysTF[lapply(sysTF,length)>0] #remove nulls
+
+  return(sysTF)
+}
+
+
+#' Helper function to make binary presence absence matrix
+#'
+#' @param sysTF output of getsysTFdata()
```

Comment:
In your roxygen headers you should add the type of the variable for your param and returns.